src/container/srv_container.c

```diff
@@ -549,16 +549,20 @@ cont_create(struct rdb_tx *tx, struct ds_pool_hdl *pool_hdl,
 
 	/* Verify the pool handle capabilities. */
 	if (!(pool_hdl->sph_capas & DAOS_PC_RW) &&
-	    !(pool_hdl->sph_capas & DAOS_PC_EX))
+	    !(pool_hdl->sph_capas & DAOS_PC_EX)) {
+		D_ERROR(DF_CONT": rpc=%p pool handle capabilities check\n",
+			DP_CONT(pool_hdl->sph_pool->sp_uuid,
+				in->cci_op.ci_uuid), rpc);
 		D_GOTO(out, rc = -DER_NO_PERM);
+	}
 
 	/* Check if a container with this UUID already exists. */
 	d_iov_set(&key, in->cci_op.ci_uuid, sizeof(uuid_t));
 	d_iov_set(&value, NULL /* buf */, 0 /* size */);
 	rc = rdb_tx_lookup(tx, &svc->cs_conts, &key, &value);
 	if (rc != -DER_NONEXIST) {
 		if (rc == 0)
-			D_DEBUG(DF_DSMS, DF_CONT": container already exists\n",
+			D_ERROR(DF_CONT": container already exists\n",
```

Comment:
Did you intend to do the following instead?

    if (rc != -DER_NONEXIST) {
        if (rc == 0)
            D_DEBUG(...);
        else
            D_ERROR(...);
        D_GOTO(out, rc);
    }